fvm/programs/programs.go

```diff
@@ -27,73 +27,57 @@ type ProgramEntry struct {
 	State    *state.State
 }
 
-type ProgramGetFunc func(location common.Location) (*ProgramEntry, bool)
-
-func emptyProgramGetFunc(_ common.Location) (*ProgramEntry, bool) {
-	return nil, false
-}
-
 // Programs is a cumulative cache-like storage for Programs helping speed up execution of Cadence
 // Programs don't evict elements at will, like a typical cache would, but it does it only
 // during a cleanup method, which must be called only when the Cadence execution has finished.
 // It it also fork-aware, support cheap creation of children capturing local changes.
 type Programs struct {
-	lock       sync.RWMutex
-	programs   map[common.LocationID]ProgramEntry
-	parentFunc ProgramGetFunc
```

Comment:
Was there a particular reason this was a function in the first place?

Reply:
I vaguely remember that there was a reason for this to be a func, but not in details, I'm okey we change it for now and when @m4ksio is back we would ask him.

Reply:
I think we wanted to prevent long chains of programs. Having a function gave us ability to copy entries from parents up, but I don't think we ever used it